### PR TITLE
NAS-117566 / 22.12 / Nas-117566 Add support for snapdev

### PR DESCRIPTION
**Pull request by Raz-Dva**
Added new field in create/update  dataset/zvol form (Advanced settings) in new storage
Create or update dataset/zvol. Test the new snapdev field. The default value is hidden. Field is inheritable.

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
1. In zvol form `snapdev` is not in Advanced settings.
2. I think the way it works for most fields is that it tries to inherit value first before going for the default value.

**Comment by Raz-Dva** [comment]
Fixed.

**Review by undsoft — APPROVED** [review]
Needs better name for enum, otherwise looks good.

File: src/app/enums/dataset.enum.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export enum DatasetSync {
   Inherit = 'INHERIT',
 }
 
-export enum DatasetSnapdir {
+export enum DatasetCommonOpt {
```
[the remark this answers is not in the document]
Comment (Raz-Dva) [reply src/app/enums/dataset.enum.ts]: Added new enum DatasetSnapdev.

**Review by undsoft — APPROVED** [review]
👍🏻

**Pull request merged**